### DISCUSSION TITLE
Github Container registry (ghcr.io) is not resolving <my-image>:latest

### DISCUSSION THREAD

**Discussion by EdmundsEcho** [community/community#23606] [New to GitHub]
<p>Hello,</p>
<p>I’m able to pull my packages referencing a specific version.  For instance, the following works:</p>
<pre><code class="lang-yaml">        image: ghcr.io/.../frontend:1642627011
        imagePullPolicy: IfNotPresent
</code></pre>
<p>However, the following does not:</p>
<pre><code class="lang-yaml">        image: ghcr.io/lucivia/frontend:latest
        imagePullPolicy: Always
</code></pre>
<p>The kubernetes cluster reports “failed to resolve”</p>
<pre><code class="lang-auto">Events:
  Type     Reason     Age                  From               Message
  ----     ------     ----                 ----               -------
  Normal   Scheduled  2m47s                default-scheduler  Successfully assigned ingress-nginx/frontend-deployment-dc8574f9b-6fdqf to xxx-control-plane
  Normal   Pulling    88s (x4 over 2m47s)  kubelet            Pulling image "ghcr.io/.../frontend:latest"
  Warning  Failed     87s (x4 over 2m47s)  kubelet            Failed to pull image "ghcr.io/.../frontend:latest": rpc error: code = NotFound desc = failed to pull and unpack image "ghcr.io/.../frontend:latest": failed to resolve reference "ghcr.io/.../frontend:latest": ghcr.io/.../frontend:latest: not found
  Warning  Failed     87s (x4 over 2m47s)  kubelet            Error: ErrImagePull
  Warning  Failed     61s (x6 over 2m46s)  kubelet            Error: ImagePullBackOff
  Normal   BackOff    47s (x7 over 2m46s)  kubelet            Back-off pulling image "ghcr.io/.../frontend:latest"
</code></pre>
<p>Is the use of <code>latest</code> keyword supported by <code>ghcr</code>?</p>
<p>Thank you for anyone with some experience using the <code>latest</code> pull feature :))</p>
<p>- E</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>In my experience it works just fine. One thing to be aware of: The <code>latest</code> tag isn’t automatic, you need to push it like any other tag.</p>

**Comment by EdmundsEcho** [comment]
<p>That’s what I was missing.  For some reason, I was expecting “latest” to be a <em>feature</em> of the pull request; i.e., that it would pull the “latest” version of the package.  To your point, “latest” is just a tag used “by convention” that I need to create.</p>
<p>In order to maintain package versions, I should create a copy of the package; one with the tag, one with the whatever I’m using to track the versions (in my case $GITHUB_RUN_ID).  Otherwise, I would have to figure out a way to both add and remove the label to ensure only one has that “honor” of being the latest.</p>
<p>Is there another more conventional approach?</p>

**Comment by airtower-luna** [comment]
<p>You don’t need to remove the tag, if you push a new <code>latest</code> (or any other label) a previous one with the same name gets overwritten. What may cause issues there is if there may be multiple builds that push the tag running in parallel, and you can’t know which one will push last. :wink:</p>
<p>Convention for container images in general is that <code>latest</code> points at the latest stable/release version. If the image is meant to be shared with anyone it’s worth considering what that means in your context.</p>

**Comment by EdmundsEcho** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> Thank you.  That is helpful and completes the understanding required to get the job done.  - E</p>
